doc/recognition.xml

```diff
@@ -296,6 +305,37 @@ The following attributes are defined for recognition info records:
 <#Include Label="StdPresentation">
 <#Include Label="methodsforfactor">
 
+<ManSection>
+<Attr Name="validatehomominput" Arg="ri, x"/>
+<Description>
+    The value of this attribute is a function of two variables. This function returns
+    a boolean which is <K>false</K> when <A>x</A> is not
+    in the source of the <Ref Attr="Homom"/> of <A>ri</A>. For example,
```

Comment:
This does not mention when another boolean is returned -- presumably only `true` is meant here (`fail` is technically also a boolean).
Anyway, technically, deciding whether an element is in the "source" may impossible at this point because in order to decide this, we'd need to have fully recognized the group. But the current language seems to require that `false` *must* be returned in this case, which is just not feasible.

Instead, I would described what it is *used* for, and then only described typical reasons for when to return true and when false later on. Maybe like this:

```
The value of this attribute, if there is any, must be a function with
two arguments: a recognition record <A>ri</A>, and an element <A>x</A>.
The function must return a boolean. If it returns <K>false</K>, then
this means that <A>x</A> cannot be in the source of the homomorphism
returned by <Ref Attr="Homom"/>. If <K>true</K> is returned, then
<A>x</A> may be contained in the source of that homomorphism, and it
must be safe to pass it to that homomorphism. I.e., that homomorphism
then must not produce an error; and if <A>x</A> turns out to be not in
the source after all, it must return <K>fail</K>.
```

Then you can go on with the example of a matrix group.

Reply:
My reading of GAP's documentation is that the things in 'source' are exactly the values which it is, in general, legal to pass to a homomorphism.

It's not required that things in the image are in the group we are identifying, the image of the homomorphism will be in general anything which satisfies the "property" of this node (has a block system, fixes some points, whatever).

However, it does seem this point is a point of confusion.